components/DatasetsTable/filters/LabelFilter.tsx

```diff
@@ -0,0 +1,62 @@
+import { useLayoutEffect, useMemo, useState } from 'react';
+
+import { css } from '@emotion/react';
+import { debounce, TextField } from '@material-ui/core';
+
+import { filterSize } from './constants';
+
+export interface LabelFilterProps {
+  /**
+   * Current label as string (what user typed).
+   */
+  label: string;
+  /**
+   * Function to set label.
+   */
+  setLabel: (label: string) => void;
+}
+
+/**
+ * Component which adjusts filtering of datasets according to label.
+ */
+export const LabelFilter = ({ label, setLabel }: LabelFilterProps) => {
+  // Since we are using debounce, value in the 'label' does not update in real time.
+  // That is why we use this state variable which displays the provided text in real time.
+  const [displayLabel, setDisplayLabel] = useState(label);
+
+  const onChangeDebounced = useMemo(() => {
```

Comment:
Nothing really "wrong" here but I usually only use memo/callback hooks when fixing a performance issue or render bug. If that's not what you're trying to do with it here then it seems unnecessary. If so disgegard this.